ra/registration-authority.go

```diff
@@ -110,6 +111,12 @@ func validateEmail(ctx context.Context, address string, resolver bdns.DNSResolve
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
+	if len(splitEmail) > 2 {
```

Comment:
This only checks if there are multiple `@` symbols in the address (something I'm not 100% sure is disallowed by the extremely complicated email spec), the real problem of #1323 is that we throw away the result of `mail.ParseAddress` which will parse a string with multiple into a single address without throwing an error.

Instead we should replace `mail.ParseAddress` with `mail.ParseAddressList` and check that the returned list of addresses only contains one address.